src/KeyedData/KeyedDataArray.php

```diff
@@ -2,54 +2,39 @@
 
 declare(strict_types=1);
 
-namespace Focus\Data;
+namespace Focus\Data\KeyedData;
 
 use ArrayAccess;
-use InvalidArgumentException;
+use ArrayObject;
+use Focus\Data\Data;
 use JmesPath\Env as JmesPath;
 use RuntimeException;
-use stdClass;
 
 use function array_key_exists;
 use function explode;
 use function gettype;
 use function is_array;
-use function is_object;
 use function preg_match;
-use function property_exists;
 use function trim;
 use function vsprintf;
 
-final readonly class KeyedData implements Data
+final readonly class KeyedDataArray implements Data
 {
-    public static function from(array|object $value): self
-    {
-        return new self($value);
-    }
-
-    public static function tryFrom(mixed $value): self
+    public static function tryFrom(mixed $value): KeyedDataArray
     {
         if ($value === null) {
             return new self();
         }
 
-        if (is_array($value) || is_object($value)) {
-            return self::from($value);
-        }
-
-        throw new InvalidArgumentException(
-            message: vsprintf(format: 'Cannot create KeyedData from %s', values: [
-                gettype($value),
-            ]),
-        );
+        return KeyedDataFactory::from($value);
     }
 
     public function __construct(
-        private array|object $value = new stdClass(),
+        private array|ArrayObject $value = [],
```

Comment:
```suggestion
        private array|ArrayAccess $value = [],
```

We should allow any `ArrayAccess` implementation.